src/Computation/PooledContext.cs

```diff
@@ -27,20 +27,20 @@ public void Dispose()
         Contexts.Add(_context);
     }
 
-    public void Assert(IEnumerable<BoolExpr> assertions)
+    public void Assert(IEnumerable<IValue> assertions)
     {
-        _solver.Add(assertions);
+        _solver.Add(assertions.Select(a => a.AsBool(this)));
```

Comment:
So we do away with my `Assertions` type now and just do the Dispose loop here?

Reply:
Yeah, either use the `Add` method on the solver one assertion at a time in a loop or do them all at once then dispose them all at once. Probably the former since I'm pretty sure there is nothing more efficient going on in there.